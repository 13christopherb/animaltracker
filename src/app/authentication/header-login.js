import React, {Component} from 'react';
import {Field, reduxForm} from 'redux-form'
import {authenticatinActions} from "./ducks/index";
import {Input} from "../form-fields/index";
import {validate} from "./validators";


/**
 * Container component for a form to create a new animal
 */
class HeaderLogin extends Component {

    constructor(props) {
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(values) {
        const user = {
            username: values.username,
            password: values.password
        };
        this.props.dispatch(authenticatinActions.login(user));
    }


    render() {
        return (
            <div>
                <form className="form-inline" onSubmit={this.props.handleSubmit(this.handleSubmit)}>
                    <Field
                        component={Input}
                        id="username"
                        name="username"
                        type="text"
                        onChange={this.props.handleChange}
                        value={this.props.value}
                        displayErrorMessage={false}
                    />
                    <Field
                        component={Input}
                        id="password"
                        name="password"
                        type="password"
                        onChange={this.props.handleChange}
                        value={this.props.value}
                        displayErrorMessage={false}
                    />
                    <button className="btn btn-outline-success my-2 my-sm-0" type="submit">Sign in</button>
                </form>
            </div>
        );
    }
}

export default reduxForm({
    form: 'HeaderLogin',
    validate
})(HeaderLogin);