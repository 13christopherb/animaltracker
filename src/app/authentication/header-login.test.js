import React from 'react';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer'
import {Provider} from 'react-redux'
import {reduxForm} from 'redux-form'
import {HeaderLogin} from './header-login';

const mockStore = configureMockStore();


jest.mock('react-dom');

const onSubmit = jest.fn();
const Decorated = reduxForm({
    form: 'HeaderLogin', onSubmit: {onSubmit}
})(HeaderLogin);

describe('<HeaderLogin />', () => {
    const defaultProps = {
        pristine: false,
        valid: true,
    };
    test('HeaderLogin renders correctly', () => {
        const store = mockStore();
        const tree = renderer.create(
            <Provider store={store}>
                <Decorated
                    {...defaultProps}
                />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot()
    });
});