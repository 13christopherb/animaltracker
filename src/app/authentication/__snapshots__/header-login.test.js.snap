// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimalForm /> AnimalForm renders correctly 1`] = `
<div>
  <form
    className="form-inline"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control "
        id="username"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control "
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="password"
        value=""
      />
      
    </div>
    <button
      className="btn btn-outline-success my-2 my-sm-0"
      type="submit"
    >
      Sign in
    </button>
  </form>
</div>
`;
